File: homework/thinhbka/lesson-7-twopointer-heap/1099. Two Sum Less Than K.py
class Solution:
    def twoSumLessThanK(self, nums: List[int], k: int) -> int:
        nums.sort()
        l = 0
        r = len(nums)-1
        ans = -1
        while l < r:
            value = nums[l]+nums[r]
            if value >= k:
                r-=1
            else:
                l+=1
                ans = max(ans,value)
        return ans

File: homework/tai_nguyen/lesson10/1099_two_sum_less_than_k.ts
function twoSumLessThanK(nums: number[], k: number): number {
  nums.sort((a, b) => a - b);

  let l = 0,
    r = nums.length - 1;
  let res = -Infinity;

  while (l < r) {
    const sum = nums[l] + nums[r];
    if (sum >= k) {
      r--;
    } else {
      res = Math.max(res, sum);
      l++;
    }
  }

  return res === -Infinity ? -1 : res;
}

console.log(twoSumLessThanK([10, 20, 30], 15));
